agent/src/com/cloud/agent/dao/impl/PropertiesStorage.java

```diff
@@ -51,6 +51,9 @@ public synchronized String get(String key) {
 
     @Override
     public synchronized void persist(String key, String value) {
+        if (!loadFromFile(_file)) {
+            s_logger.warn("Failed to load changes and then write to them");
```

Comment:
There is nobody to read the values, I refactored existing methods to allow persisting passwords/passphrases for keystore. I can refactor this to throw an exception, however that changes the previous implementation as persisting changes is not mandatory, the agent reads stuff from cmdline (in systemvms) etc. and sometimes receives information from readycommand.